assignments/callbacks.js

```diff
@@ -1,34 +1,101 @@
 const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
 
-function firstItem(arr, cb) {
-  // firstItem passes the first item of the given array to the callback function.
-}
+// function firstItem(arr, cb) {
+//   // firstItem passes the first item of the given array to the callback function.
+//   return cb(arr);
+// }
 
-function getLength(arr, cb) {
+let firstItem = (arr, cb) => {
+  return cb(arr);
+};
+
+let firstItemCallBack = array => {
+  return array[0];
+};
+
+console.log(firstItem(items, firstItemCallBack));
+
+let getLengthCallBack = array => {
+  return array.length;
+};
+
+let getLength = (arr, cb) =>{
   // getLength passes the length of the array into the callback.
-}
+  return cb(arr);
+};
 
-function last(arr, cb) {
+console.log(getLength(items, getLengthCallBack));
+
+let lastCallback = array => {
+  return array[array.length -1];
+};
+
+let last = (arr, cb) => {
   // last passes the last item of the array into the callback.
-}
+  return cb(arr);
+};
+
+console.log(last(items, lastCallback));
+
+let sumNumsCallback = (a, b) => {
+  return a + b;
+};
 
-function sumNums(x, y, cb) {
+let sumNums = (x, y, cb) => {
   // sumNums adds two numbers (x, y) and passes the result to the callback.
-}
+  return cb(x, y);
+};
 
-function multiplyNums(x, y, cb) {
+console.log(sumNums(3, 4, sumNumsCallback));
+
+let multiplyNumsCallback = (a, b) => {
+  return a * b;
+};
+
+let multiplyNums = (x, y, cb) => {
   // multiplyNums multiplies two numbers and passes the result to the callback.
-}
+  return cb(x, y);
+};
+
+console.log(multiplyNums(3, 4, multiplyNumsCallback));
+
+let containsCallback = (item, list) =>  {
+  for(let i = 0; i < list.length; i ++) {
+    if(list[i] === item) {
+      return true;
+    }
+  }
+  return false;
+};
 
-function contains(item, list, cb) {
+let contains = (item, list, cb) =>{
   // contains checks if an item is present inside of the given array/list.
   // Pass true to the callback if it is, otherwise pass false.
-}
+  return cb(item, list);
+};
+
+console.log(contains("Gum", items, containsCallback));
 
 /* STRETCH PROBLEM */
 
-function removeDuplicates(array, cb) {
+let removeDuplicatesCallback = array => {
+  let nonDuplicates = [];
+  for(let i = 0; i < array.length; i ++) {
+    if(!nonDuplicates.includes(array[i])) {
+      nonDuplicates.push(array[i]);
+    }
+  }
+  return nonDuplicates;
+};
+
+let test = [1, 2, 5, 2, 6, 1, 3, 7, 1];
+
+let removeDuplicates = (array, cb) => {
```

Comment:
If you move your code from `removeDuplicatesCallback` into the `removeDuplicates` function, and then pass `nonDuplicates` to the callback in your return statement instead of `array`, this should be good to go :+1: